Add useRFID to evseWifi config

BigKid1973 opened the pull request:
- if set to true the evse should be set to "normal mode" with api enabled
- in "rfid" mode the evse will be considered "leading" and evcc is "just" power control
- charging needs to be started at the evse using rfid or evse webgui
- evvc will not initiate a charge by itself
- evcc will continue an interrupted charge if a charge was sucessfully started until the car is disconnected
- everything else should be business as usual (pv mode should work as expected etc.)

BigKid1973 committed:
- Added userfid ( use rfid) to config for evse
- if set to true the evse should be set to "normal mode" with api enabled
- in "rfid" mode the evse will be considered "leading" and evcc is "just" power control
- charging needs to be started at the evse using rfid or evse webgui
- evvc will not initiate a charge by itself
- evcc will continue an interrupted charge if a charge was sucessfully started until the car is disconnected
- everything else should be business as usual (pv mode should work as expected etc.)

andig commented:
> evvc will not initiate a charge by itself

D.h. als Anwender kann ich niemals auf "start" drücken wenn nicht vorher jemand einen Token drangehalten hat? Ist das nicht unpraktisch?

BigKid1973 commented:
> > evvc will not initiate a charge by itself
> 
> D.h. als Anwender kann ich niemals auf "start" drücken wenn nicht vorher jemand einen Token drangehalten hat? Ist das nicht unpraktisch?
Der Modus ist ja für Leute gedacht, die mit RFIDs arbeiten wollen und bei denen die Interaktion mit der EVSE (bzw. smartWB) passiert und für die EVCC der Ladecontroller im Hintergrund ist, der nur dafür sorgt dass nicht mehr geladen wird als Strom vom Dach kommt... Auto wird eingesteckt, Charge mit RFID gestartet und dann wird nur Solarstrom geladen... Wenn keiner kommt wird pausiert, danach gehts weiter (das sollte so gehen...) ... Du kannst auch auf dem Interface des EVSE einfach auf Activate drucken... Das geht Alternativ auch...
Du kannst auch im Interface von EVCC die Modi durchschalten... Das geht alles... und wenn einmal ein Ladevorgang von der EVSE "autorisiert" wurde dann wirkt das auch sofort... Dann kannst du auch auf der EVCC stoppen und wieder starten (hoffe ich) - solange bis du das Auto abhängst... Wird dann erneut angesteckt dann muss wieder von der EVSE gestartet werden...
Ich kann da nix unpraktisches finden... Wenn man denn mit RFIDs Ladevogänge zuordnen können will, dann muss halt sichergestellt sein, dass auch RFID gestartet wird... Einzige Ausnahme ist das Webinterface für das man ein PW braucht... Dann steht im LOG halt "GUI" ... Bei der Zielgruppe wird das aber eher keiner machen... ausser dem "Admin"...

BigKid1973 commented:
Die Kommentare sind  zum Teil mist :( da hab ich geschlampt...  Da geh ich drüber... sorry...

BigKid1973 committed:
- reworked enable for rfid mode
- comments reworked

BigKid1973 committed:
- reworked disabling and setCurrent:
In RFID mode do not set current < 6 while evse is locked
With current < 6 it would not be possible to initiate a charge at the evse itself

BigKid1973 committed:
fix for MaxCurrent

andig commented:
> Wird dann erneut angesteckt dann muss wieder von der EVSE gestartet werden...

Am Ende ist das Gefrickel über 2 Systeme. Schöner wäre doch wenn ich alles über EVCC steuern könnte. An der WB selbst sollte gar nichts mehr passieren. Ich verlagere die inhaltliche Diskussion nochmal zu Slack- mehr Unterstützer.

BigKid1973 commented:
Klar... Ich habe mir jetzt nur nicht zugetraut das auf "weiter oben" im Code zu regeln...
Es gäbe aus meiner Sicht noch die Möglichkeit über die Log-Einträge zu gehen...
Und zu versuchen wirklich nur die Current zu regeln... Das Problem scheint aber zu sein: Wenn man Current auf 0 setzt, dann reagiert die EVSE wieder nicht auf den RFID...
Und fakt ist doch: Es sollte auf jeden Fall auch ohne Interaktion mit dem Web-Interface von EVCC gehen...
Wenn ein User kommt, einsteckt und sein RFID drannhebt, dann erwartet er dass danach geladen wird (zumindest sobald genug Sonne da ist, oder in EVCC der Modus auf Sofort steht) und er nicht, wenn genug Sonne da ist nochmal runter spurten muss...

BigKid1973 commented:
Ich zieh den request erstmal zurück...
Eine saubere Lösung ist nur möglich wenn evse was tut... dass die Evse im normalmode den Ladevorgang abbricht wenn man den Current auf 0 setzt ist imho fast schon ein Bug...
